debug_toolbar/middleware.py

```diff
@@ -112,7 +112,10 @@ async def __acall__(self, request):
 
         # Activate instrumentation ie. monkey-patch.
         for panel in toolbar.enabled_panels:
-            panel.enable_instrumentation()
+            if panel.panel_id == "SQLPanel":
+                await sync_to_async(panel.enable_instrumentation)()
```

Comment:
Is it possible to do this within the `SQLPanel.enable_instrumentation` itself?

Reply:
It's a bit odd to see specific logic for a single panel in the middleware. It'll be easier to understand the middleware if we don't have specific panel logic within it. It's also easier to explain why we do that in the panel and have it be in the place a person is most likely to be looking for it.

Reply:
Oh, I like that.

Reply:
we only have one panel as of now that would utilise this pattern, though it seems really intuitive.

Reply:
yep but if a people create a third party panel he need to change the code of the core for make work. That's not a great thing.